### ISSUE TITLE
SAE k1_gate test: Softmax K=1 built with 0-column logits violates the K-column SaeAssignment construction convention

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#841]
## Summary

`src/terms/sae_manifold.rs::tests::k1_gate_modes_do_not_pin_assignment_to_one` fails to construct the Softmax K=1 sub-case:

```
called `Result::unwrap()` on an `Err` value:
"SaeAssignment::new: coords length 1 must equal K=0"
```

Reproduced on committed main in Rust CI run 27084917256.

## Root cause (file:line)

The test builds the Softmax K=1 assignment with a **zero-column** logits matrix and one coord block (`src/terms/sae_manifold.rs:8935`):

```rust
let sm = SaeAssignment::from_blocks_with_mode(
    Array2::<f64>::zeros((1, 0)),   // 0 logit columns
    vec![array![[0.0]]],            // 1 coord block
    AssignmentMode::softmax(1.0),
);
```

`SaeAssignment::with_mode` validates `coords.len() == logits.ncols()` for **all** modes (`src/terms/sae_manifold.rs:2236`):

```rust
let k = logits.ncols();
if coords.len() != k {
    return Err(format!("SaeAssignment::new: coords length {} must equal K={k}", coords.len()));
}
```

With 0 logit columns, `k = 0`, so the 1-coord vector is rejected. But everywhere else in the codebase the Softmax logits matrix carries **K columns** (one per atom), not `assignment_coord_dim() = K-1`:

- `k_atoms()` is defined as `logits.ncols()` (`sae_manifold.rs:2267`).
- Softmax K=1 is constructed as `(n, 1)` logits + 1 coord at `sae_manifold.rs:9249`, and Softmax K=2 as `(n, 2)` logits + 2 coords at `sae_manifold.rs:12174` — both pass validation and fit.
- `try_assignments_row`'s K=1-softmax pin checks `self.k_atoms() == 1` (`sae_manifold.rs:8334`), i.e. it expects **1** logit column for K=1, not 0.

So the established Softmax column convention is **K columns** (canonicalized so the last reference column is 0; see `canonicalize_softmax_logits`, `sae_manifold.rs:8118`). The test's 0-column construction contradicts that convention.

## This is a test bug, not a math bug

The intended Softmax K=1 assignment is representable as `Array2::<f64>::zeros((1, 1))` + one coord (matching `k_atoms()==1`, which then fires the pin to `1.0` at `try_assignments_row`). The `(1, 0)` form encodes the `assignment_coord_dim` (K-1) layout, which the constructor does not use as its column count for any mode. The IBP-MAP and JumpReLU sub-cases in the same test use `array![[...]]` (1 column, K=1) and pass; only the Softmax sub-case uses the 0-column form.

Fix is in the test: build the Softmax K=1 case with a single-column zero logits matrix.
